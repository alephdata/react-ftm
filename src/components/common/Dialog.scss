.Dialog {
  padding-bottom: 0 !important;

  &.large {
    width: 800px;
  }

  &__content {
    position: relative;
    padding-bottom: 20px;

    &.isProcessing {
      pointer-events: none;
    }
  }

  &__loading-overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: rgba(white, 0.7);
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;

    .bp4-spinner {
      top: 50%;
      position: relative;
      transform: translateY(-50%);
    }
  }
}
