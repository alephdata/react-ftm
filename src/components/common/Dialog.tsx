import * as React from 'react'
import { Dialog as Bp3Dialog, IconName, Spinner } from '@blueprintjs/core'
import c from 'classnames';

import "./Dialog.scss";

interface IDialogProps {
  icon: IconName,
  isOpen: boolean,
  isProcessing: boolean,
  title: string,
  onClose: () => any,
  className?: string,
}

export class Dialog extends React.Component<IDialogProps> {
  render() {
    const { children, className, isProcessing, ...rest } = this.props;

    return (
      <Bp3Dialog
        className={c('Dialog', className)}
        portalClassName="dialog-portal-container"
        {...rest}
      >
        <div className={c('Dialog__content', isProcessing)}>
          {children}
          {isProcessing && (
            <div className="Dialog__loading-overlay">
              <Spinner />
            </div>
          )}
        </div>
      </Bp3Dialog>
    );
  }
}
