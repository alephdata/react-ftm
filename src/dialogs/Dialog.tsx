import * as React from 'react'
import { Dialog as Bp3Dialog, IconName, Spinner } from '@blueprintjs/core'
import { Schema } from '@alephdata/followthemoney'
import c from 'classnames';

import "./Dialog.scss";

interface IDialogProps {
  icon: IconName,
  isOpen: boolean,
  isProcessing: boolean,
  title: string,
  toggleDialog: () => any,
}

class Dialog extends React.Component<IDialogProps> {
  render() {
    const { children, icon, isOpen, isProcessing, title, toggleDialog } = this.props;

    return (
      <Bp3Dialog
        icon={icon}
        isOpen={isOpen}
        title={title}
        onClose={toggleDialog}
        className="Dialog"
        portalClassName="dialog-portal-container"
      >
        <div className={c('Dialog__content', isProcessing)}>
          {children}
          {isProcessing && (
            <div className="Dialog__loading-overlay">
              <Spinner />
            </div>
          )}
        </div>
      </Bp3Dialog>
    );
  }
}

export default Dialog;
