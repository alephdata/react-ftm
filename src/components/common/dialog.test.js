import React from 'react';
import { shallow } from 'enzyme';
import { Dialog } from './Dialog';

describe('<Dialog />', () => {
    it('Should render correctly with no properties', () => {
        const component = shallow(<Dialog />);
        expect(component).toMatchSnapshot();
    });

    it('Should render correctly with properties', () => {
        const component = shallow(<Dialog className="my-dialog" />);
        expect(component).toMatchSnapshot();
    });
});
