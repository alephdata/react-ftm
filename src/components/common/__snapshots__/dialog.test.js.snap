// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Should render correctly with no properties 1`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="Dialog"
  isOpen={false}
  portalClassName="dialog-portal-container"
>
  <div
    className="Dialog__content"
  />
</Blueprint3.Dialog>
`;

exports[`<Dialog /> Should render correctly with properties 1`] = `
<Blueprint3.Dialog
  canOutsideClickClose={true}
  className="Dialog my-dialog"
  isOpen={false}
  portalClassName="dialog-portal-container"
>
  <div
    className="Dialog__content"
  />
</Blueprint3.Dialog>
`;
